### DISCUSSION TITLE
Unusual error when running gseapy.prerank

### DISCUSSION THREAD

**Discussion by hbaker2** [zqfang/GSEApy#187] [Q&A]
Hello, I am analyzing single cell RNA sequencing data using GSEApy and am getting an unusual error when trying to run:

`res = gp.prerank(rnk=gene_rank, gene_sets='GO_Biological_Process_2021')`

**KeyError**: "'de novo' posttranslational protein folding (GO:0051084)"

Even when I change it to a different gene set, I get a different weird error:

`res = gp.prerank(rnk=gene_rank, gene_sets='KEGG_2016')`

**KeyError**: 'Glycosaminoglycan biosynthesis - chondroitin sulfate / dermatan sulfate Homo sapiens hsa00532'

My gene_rank list was generated fine though, that part worked. Your help is much appreciated.

**Comment by zqfang** [comment]
Could you send me the full trackback error message? I can't help without more context

**Reply by hbaker2** [reply]
Here is the entire output, your help is much appreciated:

2023-01-09 14:58:18,205 [WARNING] Input gene rankings contains NA values(gene name and ranking value), drop them all!
2023-01-09 14:58:18,298 [WARNING] Duplicated values found in preranked stats: 24.72% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
2023-01-09 14:58:18,388 [ERROR] No gene sets passed through filtering condition !!! 
Hint 1: Try to lower min_size or increase max_size !
Hint 2: Check gene symbols are identifiable to your gmt input.
Hint 3: Gene symbols curated in Enrichr web services are all upcases.

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_5408\3554385580.py in <module>
----> 1 res = gp.prerank(rnk=gene_rank, gene_sets='GO_Biological_Process_2021')

~\anaconda3\lib\site-packages\gseapy\__init__.py in prerank(rnk, gene_sets, outdir, pheno_pos, pheno_neg, min_size, max_size, permutation_num, weighted_score_type, ascending, threads, figsize, format, graph_num, no_plot, seed, verbose, *arg, **kwarg)
    355         verbose,
    356     )
--> 357     pre.run()
    358     return pre
    359 

~\anaconda3\lib\site-packages\gseapy\gsea.py in run(self)
    424         self._logger.info("Parsing data files for GSEA.............................")
    425         # filtering out gene sets and build gene sets dictionary
--> 426         gmt = self.load_gmt(gene_list=dat2.index.values, gmt=self.gene_sets)
    427         self.gmt = gmt
    428         self._logger.info(

~\anaconda3\lib\site-packages\gseapy\base.py in load_gmt(self, gene_list, gmt)
    233             )
    234             self._logger.error(msg)
--> 235             dict_head = "{ %s: [%s]}" % (subsets[0], genesets_dict[subsets[0]])
    236             self._logger.error(
    237                 "The first entry of your gene_sets (gmt) look like this : %s"

KeyError: "'de novo' posttranslational protein folding (GO:0051084)"

**Reply by zqfang** [reply]
It complains the gene identifiers are not matched to the GMT file. Are your gene_rank's gene_symols all upcases ?

**Reply by hbaker2** [reply]
Thank you, it worked once I capitalized the gene names!
